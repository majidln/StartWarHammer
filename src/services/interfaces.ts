export interface Ingredient {
  carbs?: Number;
  fat?: Number;
  protein?: Number;
  fiber?: Number;
  calories?: Number;
}

export interface Nutrition {
  values: Ingredient;
  percentages: Ingredient;
}

export interface Tag {
  id: String;
  type: String;
  title: String;
}

export interface Instruction {
  title: String;
  steps: Array<String>;
  header?: Object;
  footer?: Object;
}

export interface Images {
  hz: String;
  vt: String;
  brightness: String;
}

export interface Recipe {
  id: string;
  slug: String;
  title: string;
  description: string;
  nutrition: Nutrition;
  time: Object;
  difficulty: Object;
  images: Images;
  tags: Array<Tag>;
  instructionSections: Array<Instruction>;
  rating: Number;
  isMembersOnly: Boolean;
  modifiedAt: String;
}

export interface RecipePage {
  listRecipes: Array<Recipe>;
  totalSize: Number;
  nextPage: Number;
}

export interface QueryResponse {
  data: RecipePage;
  loading: Boolean;
  fetchMore: Function;
}
