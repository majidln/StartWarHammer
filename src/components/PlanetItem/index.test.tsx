import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, waitFor, fireEvent} from '@testing-library/react-native';
import {PlanetItem} from './index';

import {Planet} from '@services/interfaces';

const planet: Planet = {
  name: 'Alderaan',
  id: 'cj0o7m30ms0gk01726hgbrd0s',
  population: 2000000000,
  gravity: '1 standard',
  __typename: 'Planet',
};

it('Planet Item Snapshot renders correctly', () => {
  const tree = renderer.create(<PlanetItem planet={planet} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Planet Item render in list correctly', async () => {
  const {findByTestId} = render(<PlanetItem planet={planet} />);

  await waitFor(() => findByTestId('planetItem'));
});
