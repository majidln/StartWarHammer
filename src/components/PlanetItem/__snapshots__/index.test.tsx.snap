// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet Item Snapshot renders correctly 1`] = `
<View
  testID="planetItem"
>
  <Text>
    In PlantItem
  </Text>
  <Text>
    {"name":"Alderaan","id":"cj0o7m30ms0gk01726hgbrd0s","population":2000000000,"gravity":"1 standard","__typename":"Planet"}
  </Text>
</View>
`;
